pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1268,7 +1268,17 @@ class Timedelta(_Timedelta):
 
             kwargs = {key: _to_py_int_float(kwargs[key]) for key in kwargs}
 
-            nano = convert_to_timedelta64(kwargs.pop('nanoseconds', 0), 'ns')
+            # GH43764, making sure any nanoseconds contributions from any kwarg
+            # is taken into consideration
+            nano = convert_to_timedelta64(
+                (
+                    kwargs.pop('nanoseconds', 0)
+                    + kwargs.pop('microseconds', 0) * 1000
+                    + kwargs.pop('milliseconds', 0) * 1000000
+                    + kwargs.pop('seconds', 0) * 1000000000
+                ), 'ns'
```

Comment:
because these are floats, I assume that this will lose precision near the limits. i.e. seconds = 106751 (days) * 24*3600 with a nanosecond component specified too?

Reply:
Not really sure, but I would expect that when aiming at precision the inputs should instead be coded as integers by the caller? My motivation here is only to make the instantiation consistent in that, if float inputs are allowed, then any nanosecond contributions from any supplied `kwargs` must be taken into consideration

Reply:
indeed, if any of these components is a float.

so `pd.Timedelta(seconds=float(106751 * 24 * 3600), nanoseconds=1)`

gives `Timedelta('106751 days 00:00:00')` whereas on master we get `Timedelta('106751 days 00:00:00.000000001')`

Reply:
Could you check my last commit? It produces the same output as in `master`.